backend/benefit/applications/tests/test_applications_api.py

```diff
@@ -1216,21 +1220,105 @@ def test_attachment_delete(request, api_client, application, status, expected_co
 
 
 @pytest.mark.parametrize(
-    "status",
+    "status,upload_result",
     [
-        ApplicationStatus.DRAFT,
-        ApplicationStatus.ADDITIONAL_INFORMATION_NEEDED,
+        (ApplicationStatus.DRAFT, 201),
+        (ApplicationStatus.ADDITIONAL_INFORMATION_NEEDED, 201),
+        (ApplicationStatus.HANDLING, 403),
+        (ApplicationStatus.ACCEPTED, 403),
+        (ApplicationStatus.REJECTED, 403),
     ],
 )
-def test_pdf_attachment_upload(request, api_client, application, status):
+def test_pdf_attachment_upload_and_download_as_applicant(
+    request, api_client, application, status, upload_result
+):
     application.status = status
     application.save()
     response = _upload_pdf(request, api_client, application)
-    assert response.status_code == 201
+    assert response.status_code == upload_result
+    if upload_result != 201:
+        return
     assert len(application.attachments.all()) == 1
     attachment = application.attachments.all().first()
     assert attachment.attachment_type == AttachmentType.EMPLOYMENT_CONTRACT
     assert attachment.attachment_file.name.endswith(".pdf")
+    response = api_client.get(get_detail_url(application))
+    assert len(response.data["attachments"]) == 1
+    assert response.data["attachments"][0]["attachment_file"].startswith(
+        "http://testserver/"
+    )
+    assert (
+        "handlerapplications" not in response.data["attachments"][0]["attachment_file"]
+    )
+    # url_path = urlparse(response.data['attachments'][0]['attachment_file']).path
```

Comment:
IMHO: 
 - Leaving commented out code is not good practice.
 - If the code is not needed then it should be deleted, if it is needed then it should not be commented.